app.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+//catalog/array of available items
+Item.catalog = [];
+Item.cycles = 0;
+
+var results = document.getElementById('results');
+// Item Constructor to generate items and put them into the catalog
+
+function Item(filepath, name) {
+  this.filepath = filepath;
+  this.name = name;
+  this.views = 0;
+  this.votes = 0;
+  Item.catalog.push(this);
+}
+
+// new instances of Items
+new Item('img/bag.jpg', 'robot bag');
+new Item('img/banana.jpg', 'Banana');
+new Item('img/bathroom.jpg', 'Bathroom');
+new Item('img/boots.jpg', 'boots');
+new Item('img/breakfast.jpg', 'breakfast');
+new Item('img/bubblegum.jpg', 'bubblegum');
+new Item('img/chair.jpg', 'chair');
+new Item('img/cthulhu.jpg', 'cthulhu');
+new Item('img/dog-duck.jpg', 'dog duck');
+new Item('img/dragon.jpg', 'dragon');
+new Item('img/pen.jpg', 'pen');
+new Item('img/pet-sweep.jpg', 'pet sweep');
+new Item('img/scissors.jpg', 'scissors');
+new Item('img/shark.jpg', 'shark');
+new Item('img/sweep.png', 'sweep');
+new Item('img/tauntaun.jpg', 'tauntaun');
+new Item('img/unicorn.jpg', 'unicorn');
+new Item('img/usb.gif', 'usb');
+new Item('img/water-can.jpg', 'watering can');
+new Item('img/wine-glass.jpg', 'wine glass');
+
+
+//access the element from the DOM
+
+var imgElement = document.getElementById('catalogOption');
+var imgElement2 = document.getElementById('catalogOption2');
+var imgElement3 = document.getElementById('catalogOption3');
+// event listener to check for images on click
+imgElement.addEventListener('click',handleClick);
+imgElement2.addEventListener('click',handleClick);
+imgElement3.addEventListener('click',handleClick);
+
+var randomIndex = 0;
+var randomIndex2 = 0;
+var randomIndex3 = 0;
+var previousValues = [];
+
+// grander callback that'll grab clicks as well as run the random image producer
+function handleClick(event) {
+
+  var currentTarget = event.target.currentSrc.slice(46);
+
```

Comment:
This is probably the hardest way to do this ... slicing out the end of the long string that is the img src isn't dependable.  Setting the img alt to the name and simply inspecting that will be a lot more accurate and reliable.